src/lib/libast/tests/path/pathaccess.c

```diff
@@ -1,20 +1,41 @@
 #include "config_ast.h"  // IWYU pragma: keep
 #include "terror.h"
 
+const char *relative_path = "foo";
+
+void cleanup_handler(void) {
+    unlink(relative_path);
+}
+
 tmain() {
     UNUSED(argc);
     UNUSED(argv);
 
     char path[PATH_MAX];
+    char expanded_relative_path[PATH_MAX];
+    const char cwd[PATH_MAX];
 
     const char *paths = "/usr:/usr/bin:/usr/local/bin";
     pathaccess(paths, 0, 0, 0, path, PATH_MAX);
 
     if (strcmp(path, "/usr")) terror("pathaccess() failed to resolve path");
 
     paths = "this_path_does_not_exist";
-    
+
     if (pathaccess(paths, 0, 0, PATH_ABSOLUTE, path, PATH_MAX)) terror("pathaccess() returns path to a file that does not exist");
 
+    // Test for relative paths
+    if (open(relative_path, O_CREAT, 0666) < 0) {
+        terror("Failed to create test file");
+    }
+
+    atexit(cleanup_handler);
+
+    if (!getcwd(cwd, PATH_MAX)) terror("Failed to get current working directory");
```

Comment:
FWIW, it's safer to use `sizeof(cwd)` than `PATH_MAX` (same thing below). That way if someone at a later date decides to change the size of the array these statements won't need to be updated.